runtime/ms-rest-azure/index.d.ts

```diff
@@ -221,6 +221,65 @@ export interface InteractiveLoginOptions extends DeviceTokenCredentialsOptions {
 }
 
 /**
+ * The user type associated to a subscription in Azure.
+ */
+export enum UserType {
+  user,
+  servicePrincipal
+}
+
+/**
+ * The user associated to a subscription in Azure.
+ */
+export interface LinkedUser {
+  /**
+   * The user name. This could be a display name or a GUID.
+   */
+  name: string;
+  /**
+   * The user type: 'user', 'servicePrincipal'.
+   */
+  type: UserType
+}
+
+export interface LinkedSubscription {
+  /**
+   * The tenant that the subscription belongs to.
+   */
+  tenantId: string;
+  /**
+   * The user associated with the subscription. This could be a user or a serviceprincipal.
+   */
+  user: LinkedUser;
+  /**
+   * The environment name For example: AzureCloud, AzureChina, USGovernment, GermanCloud, or your own Dogfood environment
+   */
+  environmentName: string;
+  /**
+   * Display name of the subscription.
+   */
+  name: string;
+  /**
+   * The subscriptionId (usually a GUID).
+   */
+  id: string;
+  /**
+   * The authorization source of the subscription: 'RoleBased' , 'Legacy', 'Bypassed',' Direct', 'Management'. 
+   * It could also be a comma separated string containing more values 'Bypassed, Direct, Management'.
+   */
+  authorizationSource: string;
+  /**
+   * The state of the subscription. Example values: 'Enabled', 'Disabled', 'Warned', 'PastDue', 'Deleted'.
+   */
+  state: string;
+}
+
+export interface AuthResponse {
+  credentials: DeviceTokenCredentials | ApplicationTokenCredentials | UserTokenCredentials;
+  subscriptions: Array<LinkedSubscription>;
+}
+
```

Comment:
@lostintangent - This would be the shape of the result when Promise is resolved after calling the withAuthResponse() flavors of login